opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -559,12 +559,26 @@ void CpGrid::createCartesian(const std::array<int, 3>& dims,
 
 const std::array<int, 3>& CpGrid::logicalCartesianSize() const
 {
-    return current_view_data_->logical_cartesian_size_;
+    // Temporary. For a grid with LGRs, we set the logical cartesian size of the LeafGridView as the one for level 0.
+    //            Goal: CartesianIndexMapper well-defined for CpGrid LeafView with LGRs.
+    if (current_view_data_ == this-> data_.back().get()){
+        return this -> data_[0] -> logical_cartesian_size_;
+    }
+    else{
```

Comment:
Maybe we should make this future-proof (for the parallel version) and use `this -> distributed_data_[0]->...` in the else branch? Same below

Reply:
This suggestion breaks the following tests: distribution, distribution_test_parallel,  and lookupdata tests. I'm keeping data_[0] for now. I can have a deeper look at the potential modifications in other files, to make this change [data_ replaced by distributed_data_] work.